geary-papermc-tracking/src/main/kotlin/com/mineinabyss/geary/papermc/tracking/items/cache/PlayerItemCache.kt

```diff
@@ -24,19 +21,12 @@ abstract class PlayerItemCache<T>(
     /** Cache of up-to-date item references for slots. Used to avoid set calls when reference doesn't change. */
     private val cachedItems = MutableList<T?>(maxSize) { null }
 
-    /** Tracks slots of items that have one instance of a prefab per player. */
-    private val playerInstanced = PlayerInstancedItems()
-
     private fun removeEntity(slot: Int) {
         val entity = entities[slot].takeIf { it != 0uL }?.toGeary() ?: return
         logger.d("Removing ${entities[slot]} in slot $slot")
-        if (playerInstanced.hasInstance(entity)) {
-            playerInstanced.removeAnyKind(slot)
-        } else {
-            val pdc = entity.get<ItemStack>()?.itemMeta?.persistentDataContainer
-            if (pdc != null) entity.encodeComponentsTo(pdc)
-            entity.removeEntity()
-        }
+        val pdc = entity.get<ItemStack>()?.itemMeta?.persistentDataContainer
+        if (pdc != null) entity.encodeComponentsTo(pdc)
+        entity.removeEntity()
```

Comment:
couldnt this just be consolidated into
```kt
entity.get<ItemStack>()?.itemMeta?.persistentDataContainer?.let { entity.encodeComponentsTo(it) } ?: entity.removeEntity()
```
Is there any design reason to this? just speakin in general really

Reply:
I wanna keep the congnitive load down and we're doing separate things on each line so I think it's okay to have one line per step. Your example would also not remove the entity if an ItemStack was found, which we definitely always want to do :)

Reply:
true nvm i misread and thought i saw an else... but yeah sure
```kt
entity.get<ItemStack>()?.itemMeta?.persistentDataContainer?.let { entity.encodeComponentsTo(it) }
entity.removeEntity()
```